api/src/identity-access-management/domain/usecases/register-lti-platform.js

```diff
@@ -0,0 +1,150 @@
+    scope: [
+      'https://purl.imsglobal.org/spec/lti-ags/scope/score',
+      'https://purl.imsglobal.org/spec/lti-ags/scope/result.readonly',
+      'https://purl.imsglobal.org/spec/lti-ags/scope/lineitem',
+      'https://purl.imsglobal.org/spec/lti-nrps/scope/contextmembership.readonly',
+    ].join(' '),
+    'https://purl.imsglobal.org/spec/lti-tool-configuration': {
+      target_link_uri: `${baseUrl}/api/lti`,
+      domain: new URL(baseUrl).hostname,
+      description: 'Intégration avec la plateforme Pix',
+      messages: [
+        {
+          type: 'LtiDeepLinkingRequest',
+          target_link_uri: `${baseUrl}/api/lti/content-selection`,
+        },
+      ],
+      claims: ['sub', 'iss', 'name', 'family_name', 'given_name', 'email'],
+    },
+  };
+}
+
+/**
+ * @param {object} args
+ * @param {string} args.platformConfigurationUrl
+ * @param {string} args.registrationToken
+ * @param {typeof import('../../../shared/infrastructure/http-agent.js').httpAgent} args.httpAgent
+ * @param {typeof import('../../infrastructure/repositories/lti-platform-registration.repository.js').ltiPlatformRegistrationRepository} args.ltiPlatformRegistrationRepository
+ * @param {import('../../../shared/domain/services/crypto-service.js')} args.cryptoService
+ */
+
+export async function registerLtiPlatform({
+  platformConfigurationUrl,
+  registrationToken,
+  httpAgent,
+  ltiPlatformRegistrationRepository,
+  cryptoService,
+}) {
+  const { data: platformConfiguration, ...getConfigurationResponse } = await httpAgent.get({
+    url: platformConfigurationUrl,
+    headers: { Accept: 'application/json' },
+  });
+  if (!getConfigurationResponse.isSuccessful) {
+    logger.warn({ platformConfigurationUrl }, 'Could not fetch platform configuration');
+    throw new InvalidLtiPlatformRegistrationError('Could not fetch platform configuration');
+  }
+
+  const { error: platformConfigurationValidationError } =
+    platformOpenIdConfigurationSchema.validate(platformConfiguration);
+  if (platformConfigurationValidationError) {
+    logger.warn({ platformConfiguration }, 'Invalid LTI platform configuration.');
+    throw new InvalidLtiPlatformRegistrationError('Invalid LTI platform configuration', {
+      cause: platformConfigurationValidationError,
+    });
+  }
+
+  if (!config.lti.authorizedPlatforms.includes(platformConfiguration.issuer)) {
+    logger.warn({ issuer: platformConfiguration.issuer }, 'Unauthorized LTI platform issuer.');
+    throw new InvalidLtiPlatformRegistrationError('Unauthorized LTI platform issuer');
+  }
+
+  if (!platformConfigurationUrl.startsWith(platformConfiguration.issuer)) {
+    logger.warn(
+      { issuer: platformConfiguration.issuer, platformConfigurationUrl },
+      'Inconsistent LTI platform configuration URL.',
+    );
+    throw new InvalidLtiPlatformRegistrationError('Inconsistent LTI platform configuration URL');
+  }
+
+  const pixToolConfiguration = createPixToolConfiguration(config.baseUrl);
```

Comment:
Il y a déjà la notion de `origin` dispo dans les frontaux et l'API ne doit normalement pas être appelée en direct.

Et on récupère l'`origin` au niveau du controller avec le code suivant :
```javascript
const origin = getForwardedOrigin(request.headers)
```

```suggestion
  const pixToolConfiguration = createPixToolConfiguration(origin);
```

Par exemple https://github.com/1024pix/pix/blob/dev/api/src/identity-access-management/application/token/token.controller.js#L30

Reply:
On doit avoir un point d'entrée unique côté config de la plateforme (moodle) pour des accès qui vont être redirigés soit sur Pix App, soit sur Pix Orga. On trouverait ça étrange d'avoir un front qui soit le point d'entrée d'un autre front, d'où notre choix d'aller sur l'API. 
On va discuter de ça avec @romain-pix-cyber également. Ce serait peut être une bonne idée que tu sois là :)

Reply:
Je comprends. Avec intérêt et plaisir. J'en parle également à l'équipe Accès.

Reply:
Warning ajouté sur la var d'env.